partiql-lang/src/main/kotlin/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -1102,6 +1137,8 @@ internal class StaticTypeInferenceVisitorTransform(
 
             // keep all the `THEN` expr types even if the comparison doesn't succeed
             val simpleCaseType = inferCaseWhenBranches(thenExprs, simpleCase.default)
+            // TODO: if case input expression is union(null, missing), we will lose the missing type.
+            //   Figure out if this is intentional.
```

Comment:
In my mental model, in case of that the value expression is missing, I assumed the expression should return missing.

It is almost like

when(v) { // if missing, we failed at this stage without even going to the branch. 
   'case1' -> {...}
   'case2' -> {...}
   else -> {...}
}